src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -13702,11 +13765,25 @@ int Assembler::vex_prefix_and_encode(int dst_enc, int nds_enc, int src_enc, VexS
 }
 
 int Assembler::evex_prefix_and_encode_ndd(int dst_enc, int nds_enc, int src_enc, VexSimdPrefix pre, VexOpcode opc,
-                           InstructionAttr *attributes, bool no_flags) {
+                           InstructionAttr *attributes, bool no_flags, bool use_prefixq) {
+  // Demote RegRegReg instructions
+  if (!no_flags && dst_enc == nds_enc) {
```

Comment:
This could be replaced by call to is_demotable().

Reply:
Please see the updated code refactored to use is_demotable().